### PR TITLE
implement redis cluster broker

### PR DESCRIPTION
**Pull request by pinguingman**
This PR includes updates to support the redis cluster option for the List Queue broker. The changes involve the addition of the BaseRedisClusterBroker and ListQueueClusterBroker.

### CODE REVIEW

File: tests/test_broker.py [hunk 1]
```diff
@@ -96,3 +96,34 @@ async def test_list_queue_broker(
...
+
+    We create two workers that listen and send a message to them.
+    Expect only one worker to receive the same message we sent.
+    """
+
+    print(f"redis_cluster_url: {redis_cluster_url}")
+    broker = ListQueueClusterBroker(
+        url=redis_cluster_url, queue_name=uuid.uuid4().hex
+    )
+    worker1_task = asyncio.create_task(get_message(broker))
```
Comment (s3rius) [inline tests/test_broker.py:117]: Why have you created 2 tasks here? I guess that one task is enough to test if it works.

**Comment by s3rius** [comment]
To fix all lint problems, simply use `pre-commit run -a`.

File: tests/test_broker.py (hunk 1 above)
Comment (pinguingman) [reply tests/test_broker.py]: Agree.  I've updated the test to only use one task

File: taskiq_redis/redis_cluster_broker.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+_T = TypeVar("_T")
+
+
+class BaseRedisClusterBroker(AsyncBroker):
+    """Base broker that works with Redis Cluster."""
+
+    def __init__(
+        self,
+        url: str,
+        task_id_generator: Optional[Callable[[], str]] = None,
```
Comment (s3rius) [inline taskiq_redis/redis_cluster_broker.py:17]: Let's remove id generator from here. We have new builder API to set generator and result backend.

**Review by s3rius — APPROVED** [review]
The code looks great. Thank you for contributing. But i have a small comment to the init signature.

File: taskiq_redis/redis_cluster_broker.py (hunk 2 above)
Comment (pinguingman) [reply taskiq_redis/redis_cluster_broker.py]: Done

**Comment by s3rius** [comment]
Can you also remove result backend from the constructor?

**Comment by pinguingman** [comment]
Removed result backend

**Comment by s3rius** [comment]
All target python tests passed. Sorry for codecov error. I will fix it in a subsequent PR.

**Pull request merged**